tests/references/collections.py

```diff
@@ -64,153 +88,831 @@
                 ]
             },
         },
-        "_expected": [
-            {
-                "token_id": 2,
-                "probability": 0.0012800000000000005,
-                "max_trait_ic": 2.321928094887362,
-                "ic": 9.60964047443681,
-                "unique_traits": 4,
-                "rank": 1,
+        "_expected": {
+            2: 1,
+            1: 2,
+            4: 3,
+            5: 3,
+            3: 5,
+        },
+    },
+    {
+        "desc": cleandoc("""Collection with numeric traits."""),
+        "token_type": "non-fungible",
+        "tokens": {
+            1: {
+                "attributes": [
+                    {"name": "x", "value": 1, "display_type": "number"},
+                    {"name": "y", "value": 1, "display_type": "number"},
+                    {"name": "level", "value": 10, "display_type": "number"},
+                ],
             },
-            {
-                "token_id": 1,
-                "probability": 0.002560000000000001,
-                "max_trait_ic": 2.321928094887362,
-                "ic": 8.60964047443681,
-                "unique_traits": 3,
-                "rank": 2,
+            2: {
+                "attributes": [
+                    {"name": "x", "value": 1, "display_type": "number"},
+                    {"name": "y", "value": 1, "display_type": "number"},
+                    {"name": "level", "value": 9, "display_type": "number"},
+                ],
             },
-            {
-                "token_id": 4,
-                "probability": 0.04608000000000001,
-                "max_trait_ic": 2.321928094887362,
-                "ic": 4.439715472994499,
-                "unique_traits": 1,
-                "rank": 3,
+            3: {
+                "attributes": [
+                    {"name": "x", "value": 5, "display_type": "number"},
+                    {"name": "y", "value": 5, "display_type": "number"},
+                    {"name": "level", "value": 9, "display_type": "number"},
+                ],
             },
-            {
-                "token_id": 5,
-                "probability": 0.04608000000000001,
-                "max_trait_ic": 2.321928094887362,
-                "ic": 4.439715472994499,
-                "unique_traits": 1,
-                "rank": 3,
+        },
+        "_expected": {
+            3: 1,
+            1: 2,
+            2: 3,
+        },
+    },
+    {
+        "desc": cleandoc("""Collection with date traits."""),
+        "token_type": "non-fungible",
+        "tokens": {
+            1: {
+                "attributes": [
+                    {
+                        "name": "start",
+                        "value": 1640000000,
```

Comment:
Could you make a test with string formatted dates (`"2022-01-01T00:00:00"` and `"2022-01-01"`) as well? It should process those as well so knowing that it does would be great.

Reply:
Added